### ISSUE TITLE
IDISP003 does not detect disposal of previous instance copied to local

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/IDisposableAnalyzers#100]
Version: 2.0.3.1

```cs
class Repro : IDisposable
{
    private Disposable current;

    public void Test()
    {
        var old = current;
        current = new Disposable(); // IDISP003: Dispose previous before re-assigning.
        if (old != current) old.Dispose();
    }

    public void Dispose()
    {
        current?.Dispose();
    }
}
```

Alternatives:

```cs
        var @new = new Disposable();
        if (@new != current)
        {
            current.Dispose();
            current = @new;
        }
```
